util/dvsim/SimCfg.py

```diff
@@ -426,18 +461,13 @@ def _create_deploy_objects(self):
             builds.append(item)
             build_map[build] = item
 
-        runs = []
-        for test in self.run_list:
-            for num in range(test.reseed):
-                item = RunTest(num, test, self)
-                if self.build_only is False:
-                    build_map[test.build_mode].sub.append(item)
-                runs.append(item)
+        if self.build_only:
+            build_map = None
 
         self.builds = builds
-        self.runs = runs
+        self.runs = SimCfg._expand_run_list(self.run_list, build_map, self)
```

Comment:
Oh, that's much cleaner. I'll do something like that (although not quite the same: it's probably better to make sure we set the runs field to something even when `build_only` is true). The static method was just because I'd moved it into the class namespace without thinking too hard.